File: frontend/src/api/project.ts
import request from '../utils/request'

export interface Project {
  id: number
  user_id: number
  name: string
  description: string
  created_at: string
}

export function getProjects(): Promise<Project[]> {
  return request.get('/projects')
}

export function getProject(id: number): Promise<Project> {
  return request.get(`/projects/${id}`)
}

export function createProject(data: { name: string; description: string }): Promise<Project> {
  return request.post('/projects', data)
}

export function updateProject(id: number, data: { name: string; description: string }): Promise<Project> {
  return request.put(`/projects/${id}`, data)
}

export function deleteProject(id: number) {
  return request.delete(`/projects/${id}`)
}

File: backend/internal/model/project.go
package model

import "time"

type Project struct {
	ID          uint      `gorm:"primaryKey" json:"id"`
	UserID      uint      `gorm:"index;not null" json:"user_id"`
	Name        string    `gorm:"size:100;not null" json:"name"`
	Description string    `gorm:"size:500" json:"description"`
	CreatedAt   time.Time `json:"created_at"`
}
